

If we list all the natural numbers below 10 that are multiples of 3 or 5, we get 3, 5, 6 and 9. The sum of these multiples is 23.

Find the sum of all the multiples of 3 or 5 below 1000.


In [2]:
i=1
sum=0
while(i<1000):
    if i%5==0 or i%3==0:
        sum+=i
    i=i+1
print('the sum is : ',sum)

the sum is :  233168


It was an easy out experience to solve this. but to elaise that this was a certain simple problem which i used for loop which made the program complexity n . It is possible to complete this with least complexity. It would be to craft equation for sum of numbers with 3 and 5 divisible. but that too is complex we need to find if any of these have common to remove recurrance. 
hence a simple mathematical solution should be found out.

*Solve by hand using an adaptation of Gauss' method for adding 1 through 100 -- i.e., 1 + 2 + ... + 100 = (100 + (100 + 1))/2. That is, summing 1 through 100 is like taking (100 + 1) + (99 + 2) + ... + (51 + 50), which is 101*50. 

In this case, the sum of all natural numbers less than 1000 and divisible by 3 is equal to (333 * (999 + 3))/2. That is, 
the highest number divisible by 3 under 1000 is 999, 
and 999/3 = 333. We have (999 + 3) + (996 + 6) + ... + (504 + 498) + 501, 
which is equivalent to 1002*(333/2) = 1002*166.5 
(because 333 is odd, we will have 501 right in the middle, which is divisible by 3 but not paired/added with another number). 

We do similar work for all numbers divisible by 5, as well as those divisible by 15 (which will be counted twice and hence must be subtracted once): for the sum of all divisible by 5 we calculate (199 * (995 + 5))/2 and for the sum of all divisible by 15 we calculate (66 * (990 + 15))/2. Hence, for n in
[1,1000): (sum all n divisible by 3) + (sum all n divisible by 5) - (sum all n divisible by 15) = 
[(333 * (999 + 3))/2] + [(199 * (995 + 5))/2] - [(66 * (990 + 15))/2] = 233,168.*